src/deepsparse/transformers/pipelines/text_generation.py

```diff
@@ -261,15 +255,33 @@ def process_inputs(self, inputs: TextGenerationInput) -> List[numpy.ndarray]:
             truncation=truncate,
         )
 
+        if (
+            len(input_tokens[0]) < self.sequence_length
+            and not inputs.fixed_sequences_length
+        ):
+            # if the input is shorter than the sequence length,
+            # we need to pad it to the sequence length
+            padding = "max_length"
+            input_tokens = self.tokenizer(
+                inputs.sequences,
+                return_tensors="np",
+                max_length=self.sequence_length,
+                padding=padding,
+                truncation=truncate,
+            )
+
         attention_mask = input_tokens["attention_mask"]
 
-        # TODO: Positions input is not required by BLOOM
-        # let's make it optional in the future
         positions = attention_mask.cumsum(1) * attention_mask
         positions -= 1  # assert that positions start at 0
         positions_input = dict(positions=positions)
 
-        input_tokens = {**input_tokens, **positions_input}
+        causal_mask = create_causal_mask(
+            input_tokens["input_ids"], input_tokens["attention_mask"]
+        )
+        causal_mask_input = dict(causal_mask=causal_mask)
```

Comment:
nit: why do we need to wrap it into a dict, can't we just below do
```python
input_tokens = dict(**input_tokens, positions=positions, causal_mask=causal_mask)
```

Reply:
yes, what you propose is the same thing but shorter